sema/commons/j2/j2_functions.py

```
@@ -46,43 +55,78 @@ def xsd_format_boolean(content, quote, suffix):
+            "use datetime format for datetime values, or past .date() result"
+        )
+
     if not isinstance(content, date):
         asdt = parser.isoparse(content).date()
     else:
         asdt = content
     return xsd_value(asdt.isoformat(), quote, "xsd:date")
 
 
-def xsd_format_datetime(content, quote, suffix):
+def xsd_format_gyear(content: Any, quote: str, *_: Any) -> str:
```

Ruff (0.8.2)

Dynamically typed expressions (typing.Any) are disallowed in `content`

(ANN401)